src/utils.js

```diff
@@ -257,9 +257,20 @@ export function detectIEVersion() {
  * Shallow comparison between 2 arrays
  * @param  {Array} arr1
  * @param  {Array} arr2
+ * @param  {Function} cb {optional}
  * @return {Bool}
  */
-export function isEqualArray(arr1, arr2) {
+export function isEqualArray(arr1, arr2, cb) {
+  // return false if both are not an array
+  if (!Array.isArray(arr1) || !Array.isArray(arr2)) {
+    return false;
+  }
+
+  // envoke the callback function if there is any
+  if (typeof cb === 'function') {
+    return cb(arr1, arr2);
```

Comment:
This wasn't quite what I was thinking. I was thinking you could pass in a custom comparator in the `every` method. So something like:

```
function compare(arr1, arr2, fn) {
  if (typeof fn !== 'function') {
    fn = (element, index) => elenent === arr2[index];
 }
 return arr1.every(fn);
}

```

Doing it that way means you don't have to duplicate any checks if you want to override the logic